src/vs/editor/contrib/codeAction/browser/types.ts

```diff
@@ -44,6 +44,19 @@ export const enum CodeActionAutoApply {
 	Never = 'never',
 }
 
+export enum CodeMenuOpenedFrom {
```

Comment:
Also maybe this type name needs to be more generic since it is passed around a lot of places (even though the menu is the only consumer). Perhaps `CodeActionTriggerReason` or `CodeActionTriggerSource`?